packages/core/src/rules/common/operation-tag-defined.ts

```diff
@@ -11,6 +11,13 @@ export const OperationTagDefined: Oas3Rule | Oas2Rule = () => {
       definedTags = new Set((root.tags ?? []).map((t) => t.name));
     },
     Operation(operation: Oas2Operation | Oas3Operation, { report, location }: UserContext) {
+      if (!operation?.hasOwnProperty('tags')) {
+        report({
+          message: `Operation tags should be defined`,
+          location: location.child('tags').key(),
+        });
+        return;
+      }
       if (operation.tags) {
```

Comment:
Wouldn't it be better to add the `else` clause to this `if` and put the changes there? 

```suggestion
      if (operation?.tags) {
```

Reply:
i put the check first to fail fast.  the entire rule can be refactored to check for tags, then check for global tags without having two if statements.  Let me know if you want me to do that.

Reply:
updated per your suggestion